### PR TITLE
Adding missing TypeScript declaration of setFeatureToken

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Reference to the issue(s) addressed by this PR: https://github.com/realm/realm-js-private/issues/322

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -367,6 +367,8 @@ declare namespace Realm.Sync {
     function removeAllListeners(name?: string): void;
     function removeListener(regex: string, name: string, changeCallback: (changeEvent: ChangeEvent) => void): void;
     function setLogLevel(logLevel: 'all' | 'trace' | 'debug' | 'detail' | 'info' | 'warn' | 'error' | 'fatal' | 'off'): void;
+    function setFeatureToken(token: string): void;
+    // FIXME: remove setAccessToken after 2.0.
```
Comment (blagoev) [inline lib/index.d.ts]: Isn't it better to mark this as @deprecated and add an issue for 2.0 to make the relevant change against the 2.0 branch. Fixmes are forgotten by definition

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ X.Y.Z Release notes
 * Added `Realm.deleteFile` for deleting a Realm (#363).
 
 ### Bug fixes
-* Adding missing TypeScript declation (#1283).
+* Adding missing TypeScript declations; Permissions (#1283) and `setFeatureToken()`.
```
Comment (nirinchev) [inline CHANGELOG.md]: Should this be `declarations` or even better, `definitions`?

File: lib/index.d.ts (hunk 1 above)
Comment (kneth) [reply lib/index.d.ts]: I have created https://github.com/realm/realm-js/issues/1300

**Comment by blagoev** [comment]
👍

**Pull request merged**